### PR TITLE
New Test: TestTXTRegisterSpaceValid

### PR DESCRIPTION
**Pull request by ChriMarMe**
checks the boundaries of TXT Register Space below 4Gib

### CODE REVIEW

File: pkg/test/memory.go [hunk 1]
```diff
@@ -163,6 +175,47 @@ var (
...
+	}
+
+	if uint64(regs.HeapBase) >= api.FourGiB {
+		return false, fmt.Errorf("HeapBase > 4Gib")
+	}
+
+	if uint64(regs.HeapBase+regs.HeapSize) >= api.FourGiB {
+		return false, fmt.Errorf("HeapBase + HeapSize >= 4Gib")
+	}
+	if regs.HeapSize >= uint32(0xF00000) {
```
Comment (PatrickRudolph) [inline pkg/test/memory.go]: <

File: pkg/test/memory.go [hunk 2]
```diff
@@ -163,6 +175,47 @@ var (
...
+
+	if uint64(regs.SinitBase) >= api.FourGiB {
+		return false, fmt.Errorf("SinitBase >= 4Gib")
+	}
+
+	if uint64(regs.SinitBase+regs.SinitSize) >= api.FourGiB {
+		return false, fmt.Errorf("SinitBase + SinitSize >= 4Gib")
+	}
+
+	if regs.SinitSize >= uint32(0x500000) {
```
Comment (PatrickRudolph) [inline pkg/test/memory.go]: <

File: pkg/test/memory.go [hunk 3]
```diff
@@ -163,6 +175,47 @@ var (
...
+		return false, fmt.Errorf("HeapBase > 4Gib")
+	}
+
+	if uint64(regs.HeapBase+regs.HeapSize) >= api.FourGiB {
+		return false, fmt.Errorf("HeapBase + HeapSize >= 4Gib")
+	}
+	if regs.HeapSize >= uint32(0xF00000) {
+		return false, fmt.Errorf("Heap size too small")
+	}
+
```
Comment (PatrickRudolph) [inline pkg/test/memory.go]: SinitBase < HeapBase missing

File: pkg/test/memory.go [hunk 4]
```diff
@@ -14,34 +14,39 @@ var (
...
 	}
 	testhostbridgeDPRcorrect = Test{
-		Name:     "CPU DMA protected range equals hostbridge DPR",
-		Required: false,
-		function: TestHostbridgeDPRCorrect,
-		Status:   TestImplemented,
+		Name:         "CPU DMA protected range equals hostbridge DPR",
+		Required:     false,
+		function:     TestHostbridgeDPRCorrect,
+		dependencies: []*Test{&testtxtdprislocked},
```
Comment (PatrickRudolph) [inline pkg/test/memory.go]: hostbridge DPR != txt DPR

File: pkg/test/memory.go [hunk 5]
```diff
@@ -14,34 +14,39 @@ var (
...
 	}
 	testhostbridgeDPRislocked = Test{
-		Name:     "CPU hostbridge DPR register locked",
-		Required: true,
-		function: TestHostbridgeDPRisLocked,
-		Status:   TestImplemented,
+		Name:         "CPU hostbridge DPR register locked",
+		Required:     true,
+		function:     TestHostbridgeDPRisLocked,
+		dependencies: []*Test{&testhostbridgeDPRcorrect},
```
Comment (PatrickRudolph) [inline pkg/test/memory.go]: no need for that

File: pkg/test/memory.go (hunk 1 above)
Comment (ChriMarMe) [reply pkg/test/memory.go]: done

File: pkg/test/memory.go (hunk 3 above)
Comment (ChriMarMe) [reply pkg/test/memory.go]: done

File: pkg/test/memory.go (hunk 5 above)
Comment (ChriMarMe) [reply pkg/test/memory.go]: done

File: pkg/test/memory.go [hunk 6]
```diff
@@ -160,9 +170,56 @@ var (
 )
 
 var (
-	biosdata api.TXTBiosData
+	biosdata     api.TXTBiosData
+	minHeapSize  = uint32(0xF00000)
```
Comment (PatrickRudolph) [inline pkg/test/memory.go]: 0xF0000 and 0x50000

**Pull request merged**